### PR TITLE
chore: better error overlay for component testing

### PR DESCRIPTION
**Pull request by lmiller1990**
Show webpack dev server compilation errors is there are any using the established `emit('script:error')` pattern.

image

It will automatically clear the error overlay if the error is fixed and run the test. This was fairly easy due to the nice webpack HMR API by @dmtrKovalenko.

It would have been idea to have something webpack agnostic but you cannot `try/catch` for webpack errors, you must use their API. 🤷‍♂️ 

I am not sure this will work for other errors, like in `cypress/commands.js`, but this is a good start. Collecting errors from other places should be fairly easy. 

Here is a GIF showing the feature in action.

https://recordit.co/XpI8x6OlTN

### CODE REVIEW

File: packages/runner-ct/src/lib/webpack-hmr-client.ts [hunk 1]
```diff
@@ -30,12 +36,14 @@ export function connectWebpackHmr ({ url, onReload }) {
...
     },
-    errors: (errors) => {
+    errors: (errors: string[]) => {
       console.error('Errors while compiling. Reload prevented.')
 
       errors.forEach(
         (warning) => console.warn(stripAnsi(warning)),
       )
+
+      eventManager.emit('script:error', { error: errors[0] })
```
Comment (lmiller1990) [inline packages/runner-ct/src/lib/webpack-hmr-client.ts:46]: I guess showing the first error is fine. We could do `join` to show them all, but it look like `<ScriptError>` is designed to just show one error, not many.

**Comment by dmtrKovalenko** [comment]
AFAI remember cypress shows critcial error inside the reporter? The layout looks a little bit confusing

File: packages/runner-ct/src/lib/webpack-hmr-client.ts [hunk 2]
```diff
@@ -30,12 +36,14 @@ export function connectWebpackHmr ({ url, onReload }) {
 
       onReload()
     },
-    errors: (errors) => {
+    errors: (errors: string[]) => {
```
Comment (dmtrKovalenko) [inline packages/runner-ct/src/lib/webpack-hmr-client.ts:39]: 🤦‍♂️ lets enable `noImplicitAny`

Comment (lmiller1990) [reply packages/runner-ct/src/lib/webpack-hmr-client.ts:39]: We should be I think that'll need to be another PR - there are many things that need typing haha

**Comment by lmiller1990** [comment]
@dmtrKovalenko are you suggesting we should show the error in the reporter instead? We could, but what about the case of the initial compile failed (no reporter)?

Also, we already had a `<ScriptError>` component set up, is that used for something different?

File: packages/runner-ct/src/iframe/iframes.scss [hunk 3]
```diff
@@ -1,3 +1,3 @@
 .iframes-ct-container { 
-  margin-left: 16px;
+  margin: 16px;
 }
```
Comment (JessicaSachs) [inline packages/runner-ct/src/iframe/iframes.scss:3]: Why is there no linter rule for this?!?!

```suggestion
}

```

File: packages/runner-ct/src/lib/webpack-hmr-client.ts [hunk 4]
```diff
@@ -21,6 +22,11 @@ export function connectWebpackHmr ({ url, onReload }) {
     },
     'ok': () => {
       onReload()
+
+      // in the scenario a test has not been selected,
+      // just clear the error overlay to vusually indicate to the user
```
Comment (JessicaSachs) [inline packages/runner-ct/src/lib/webpack-hmr-client.ts:27]:
```suggestion
      // just clear the error overlay to visually indicate to the user
```

File: packages/runner-ct/src/lib/webpack-hmr-client.ts (hunk 1 above)
Comment (JessicaSachs) [reply packages/runner-ct/src/lib/webpack-hmr-client.ts:46]: I've seen multiple errors before (Vite??), but... this is sufficient for webpack's model. We generally only care about transpile failing and that's a single process.exit anyway

**Review by JessicaSachs — APPROVED** [review]
Typos and whitespace. Otherwise looks good to me.

**Review by dmtrKovalenko — APPROVED** [review]

**Pull request merged**